source/developers-guide/cookie-consent-manager/index.md

```diff
@@ -0,0 +1,195 @@
+
+    $collection = new CookieCollection();
+    $collection->add(new CookieStruct(
+        'comfort',
+        $pluginNamespace->get('my_cookie_label'),
+        CookieGroupStruct::COMFORT
+    ));
+
+    return $collection;
+}
+```
+
+The third parameter is optional and represents your cookie's group. If none is applied, the "Others" group is used.
+You can find all default groups as constants in the `\Shopware\Bundle\CookieBundle\Structs\CookieGroupStruct`.
+Also, have a look at the next headline to figure out how to register your cookie group, if necessary.
+
+The last line of your method then returns the `CookieCollection` - and that's it, you've succesfully registered your own cookie.
+
+## Registering an own cookie group
+
+Normally, you wouldn't want to do this, but there might be a use case forcing to create your own cookie group.
+This is just as simple as registering your own cookie.
+
+Once again, let's first start with the example, only to explain it in short then afterwards.
+
+```php
+public static function getSubscribedEvents(): array
+{
+    return [
+        'CookieCollector_Collect_Cookie_Groups' => 'addCookieGroup',
+    ];
+}
+
+public function addCookieGroup(\Enlight_Event_EventArgs $args): CookieGroupCollection
+{
+    $pluginNamespace = $this->container->get('snippets')->getNamespace('my_plugins_snippet_namespace');
+
+    $collection = new CookieGroupCollection();
+    $collection->add(new CookieGroupStruct(
+        'custom',
+        $pluginNamespace->get('custom_label'),
+        $pluginNamespace->get('custom_description')
+    ));
+
+    return $collection;
+}
+```
+
+As already mentioned, this looks very similar to registering your custom cookie.
+First of all, the used event is named `CookieCollector_Collect_Cookie_Groups` this time.
+
+It expects a `\Shopware\Bundle\CookieBundle\CookieGroupCollection` as the return, thus you need to create it in your event listener.
+It then has to be filled using a `\Shopware\Bundle\CookieBundle\Structs\CookieGroupStruct`, which accepts four parameters,
+two of them being required.
+
+The first one is the technical name of the group, which is also used by a cookie later then. Maybe add a constant for this.
+
+The second parameter is the actual label to be shown with this group. Once more, make sure to add your translations just like shown in the example
+above.
+
+The third parameter is optional and represents a description, which will additionally be shown when the group is expanded.
+
+The last parameter marks your group as "required", which will prevent any changes being made to it and it always being active.<br />
+**Do not use this, unless you're a hundred percent sure what you're doing! This might come with legal issues!**
+
+Afterwards, only return your new collection, and that's it!
+
+<div class="alert alert-warning">
+<strong>Note</strong>: Groups without a single assigned cookie are hidden by default!
```

Comment:
Hat das einen bestimmten Grund? Dann sollten wir den hier nennen.

Reply:
Nein, hat keinen speziellen Grund. Eine Gruppe dient nur dem Zweck, einfacher alle Cookies einer Gruppe zu aktivieren / deaktivieren. Wenn es aber keine Cookies in der Gruppe gibt, erfüllt sie keinen Zweck